h2o-r/tests/testdir_algos/automl/runit_automl_leaderboard.R

```diff
@@ -20,60 +21,63 @@ automl.leaderboard.test <- function() {
     model_ids <- as.vector(aml2@leaderboard$model_id)
+    # check that no excluded algos are present in leaderboard
     exclude_algo_count <- sum(sapply(exclude_algos, function(i) sum(grepl(i, model_ids))))
     expect_equal(exclude_algo_count, 0)
     include_algos <- c(setdiff(all_algos, exclude_algos), "XRT")
+    # check that expected algos are included in leaderboard
     for (a in include_algos) {
       expect_equal(sum(grepl(a, model_ids)) > 0, TRUE)
     }
 
     # Multinomial:
     fr3 <- as.h2o(iris)
-    exclude_algos <- c("DeepLearning")
+    exclude_algos <- c("XGBoost")
     aml3 <- h2o.automl(y = 5, training_frame = fr3, max_models = 10,
                        project_name = "r_lb_test_aml3",
                        exclude_algos = exclude_algos)
     aml3@leaderboard
     expect_equal(names(aml3@leaderboard),c("model_id", "mean_per_class_error", "logloss", "rmse", "mse"))
     model_ids <- as.vector(aml3@leaderboard$model_id)
+    # check that no excluded algos are present in leaderboard
     exclude_algo_count <- sum(sapply(exclude_algos, function(i) sum(grepl(i, model_ids))))
     expect_equal(exclude_algo_count, 0)
+    # check that expected algos are included in leaderboard
     include_algos <- c(setdiff(all_algos, exclude_algos), "XRT")
     for (a in include_algos) {
       expect_equal(sum(grepl(a, model_ids)) > 0, TRUE)
     }
 
-# Below fails bc there are no models in the leaderboard, but AutoML needs to check the models to get the
-# model type (binomial, multinomial, or regression)
-#    # Exclude all the algorithms, check for empty leaderboard
-#    fr1 <- h2o.uploadFile(locate("smalldata/logreg/prostate.csv"))  #need to reload data (getting error otherwise)
-#    fr1["CAPSULE"] <- as.factor(fr1["CAPSULE"])
-#    exclude_algos <- c("GLM", "DRF", "GBM", "DeepLearning", "StackedEnsemble")
-#    aml4 <- h2o.automl(y = 2, training_frame = fr1, max_runtime_secs = 5,
-#                       project_name = "r_lb_test_aml4",
-#                       exclude_algos = exclude_algos)
-#    aml4@leaderboard
-#    #expect_equal(names(aml4@leaderboard), c("model_id","auc","logloss", "mean_per_class_error", "rmse", "mse"))
-#    # TO DO: for empty leaderboards there's a dummy row for some reason.
-#    expect_equal(nrow(aml4@leaderboard), 1)
+    
+    # Exclude all the algorithms, check for empty leaderboard
+    fr1 <- h2o.uploadFile(locate("smalldata/logreg/prostate.csv"))  #need to reload data (getting error otherwise)
+    fr1["CAPSULE"] <- as.factor(fr1["CAPSULE"])
+    exclude_algos <- c("GLM", "DRF", "GBM", "DeepLearning", "XGBoost", "StackedEnsemble")
+    aml4 <- h2o.automl(y = 2, training_frame = fr1, max_runtime_secs = 5,
+                       project_name = "r_lb_test_aml4",
+                       exclude_algos = exclude_algos)
+    aml4@leaderboard
+    # there's a dummy row of NAs in an "empty" leaderboard
```

Comment:
> there's a dummy row of NAs in an "empty" leaderboard

is it something we plan to fix?

Reply:
Yes, but not high priority. We tried to fix it once before and for some reason it was not trivial, I think.